Apps/AdminWebClient/src/Server/Services/AuthenticationService.cs

```diff
@@ -107,5 +116,59 @@ public AuthenticationProperties GetAuthenticationProperties(string redirectPath)
 
             return authProps;
         }
+
+        /// <inheritdoc />
+        public void SetLastLoginDateTime()
+        {
+            ClaimsPrincipal? user = this.httpContextAccessor.HttpContext?.User;
+            AuthenticationData authData = this.GetAuthenticationData();
+            if (authData.IsAuthenticated && authData.User != null && user != null)
+            {
+                DateTime jwtAuthTime = GetAuthDateTime(user);
+
+                DBResult<Database.Models.AdminUserProfile> result = this.profileDelegate.GetAdminUserProfile(preferredUsername: authData.User.Id);
+                if (result.Status == DBStatusCode.NotFound)
+                {
+                    // Create profile
+                    Database.Models.AdminUserProfile newProfile = new()
+                    {
+                        // Keycloak id is case insensitive
+                        Username = authData.User.Id.ToLower(CultureInfo.CurrentCulture),
+                        Email = authData.User.Email,
+                        LastLoginDateTime = jwtAuthTime,
+                    };
+                    DBResult<Database.Models.AdminUserProfile> insertResult = this.profileDelegate.Add(newProfile);
+                    if (insertResult.Status == DBStatusCode.Error)
+                    {
+                        this.logger.LogError("Unable to add admin user profile to DB.... {Result}", JsonSerializer.Serialize(insertResult));
+                    }
+                }
+                else
+                {
+                    // Update profile
+                    result.Payload.LastLoginDateTime = jwtAuthTime;
+                    DBResult<Database.Models.AdminUserProfile> updateResult = this.profileDelegate.Update(result.Payload);
+                    if (updateResult.Status == DBStatusCode.Error)
+                    {
+                        this.logger.LogError("Unable to update admin user profile to DB... {Result}", JsonSerializer.Serialize(updateResult));
+                    }
+                }
+            }
+        }
+
+        private static DateTime GetAuthDateTime(ClaimsPrincipal claimsPrincipal)
+        {
+            // auth_time is not mandatory in a Bearer token.
+            string rowAuthTime = claimsPrincipal.FindFirstValue("auth_time");
+
+            // get token  "issued at time", which *is* mandatory in JWT bearer token.
+            rowAuthTime = rowAuthTime ?? claimsPrincipal.FindFirstValue("iat");
```

Comment:
Nit: the analyzers suggest this line could be simplified using the null-coalescing assignment operator.

```rowAuthTime ??= claimsPrincipal.FindFirstValue("iat");```

Reply:
Addressed.